File: src/main/Java/boj_step9/boj1978.java
import java.io.*;
import java.math.BigInteger;
import java.util.StringTokenizer;

public class boj1978 {
    public static void main(String[] args) throws IOException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(System.out));
        int n = Integer.parseInt(reader.readLine()), count = 0;
        long[] primes = new long[n];

        StringTokenizer tokenizer = new StringTokenizer(reader.readLine());
        for (int i = 0; i < primes.length; i++) {
            primes[i] = Long.parseLong(tokenizer.nextToken());
            if (BigInteger.valueOf(primes[i]).isProbablePrime(10)) count++;
        }
        writer.append(String.valueOf(count)).close();
    }
}

File: src/main/boj_step9/boj1978.kt
import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter
import kotlin.contracts.contract
import kotlin.math.sqrt

fun main() {
    val r = BufferedReader(InputStreamReader(System.`in`))
    val w = BufferedWriter(OutputStreamWriter(System.out))
    val n = r.readLine().toInt()
    val arr = r.readLine().split(" ").map { it.toInt() }
    var count = 0
    arr.forEach {
        if (it == 1) return@forEach
        val sqrt = sqrt(it.toDouble()).toInt()
        var isPrime = true
        for (i in 2 .. sqrt) if (it % i == 0) {
            isPrime = false
            break
        }
        if (isPrime) count++
    }
    w.write(count.toString())
    w.close()
}